### DISCUSSION TITLE
Running brew comes up with `gem pristine` related commands

### DISCUSSION THREAD

**Discussion by Luckey-Elijah** [Homebrew/discussions#898] [Everyday usage]
Each time I use `brew`, the command is immediately followed with:

```bash
$ brew doctor
```
```console
Ignoring commonmarker-0.17.13 because its extensions are not built. Try: gem pristine commonmarker --version 0.17.13
Ignoring eventmachine-1.2.7 because its extensions are not built. Try: gem pristine eventmachine --version 1.2.7
Ignoring http_parser.rb-0.6.0 because its extensions are not built. Try: gem pristine http_parser.rb --version 0.6.0
Ignoring nokogiri-1.10.9 because its extensions are not built. Try: gem pristine nokogiri --version 1.10.9
Ignoring sassc-2.3.0 because its extensions are not built. Try: gem pristine sassc --version 2.3.0
Ignoring unf_ext-0.0.7.6 because its extensions are not built. Try: gem pristine unf_ext --version 0.0.7.6
Your system is ready to brew.
```

--- 

I'm opening this discussion because I was able to resolve the issue by running each  the following:

```bash
$ sudo gem pristine commonmarker --version 0.17.13
$ sudo gem pristine eventmachine --version 1.2.7
$ sudo gem pristine http_parser.rb --version 0.6.0
$ sudo gem pristine nokogiri --version 1.10.9
$ sudo gem pristine sassc --version 2.3.0
$ sudo gem pristine unf_ext --version 0.0.7.6
```

I tried running the commands without `sudo` but continued to recieve errors like:
```console
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.6.0 directory.
```

Possibly someone else has a method to perform these operations as an non-`sudo`/admin environment.

Again, I opened the discussion to show others my solution. Happy brewing 😄 !

**Comment by SMillerDev** [comment]
None of these are in a clean macOS install. If you install gems system wide they can interfere with anything that runs ruby (like brew).

**Reply by Luckey-Elijah** [reply]
Do you know why gem was coming up with these messages? I'm not versed in gem or ruby.

**Reply by Bo98** [reply]
It's because we recently enabled warnings - so the messages were just hidden before.

I'm looking into tweaks to avoid using system gems, and thus prevent these warnings.

**Reply by Luckey-Elijah** [reply]
That makes a lot sense. Brew was working perfectly fine when the warnings were showing up.
